From 3821b4ad662385894e62f10197099945701d0dbf Mon Sep 17 00:00:00 2001
From: Pavan Nikhilesh <pbhagavatula@marvell.com>
Date: Tue, 8 Jan 2019 05:00:40 +0000
Subject: [PATCH 032/767] octeontx2-af: ignore sso lf count when checking
 pffunc validity

commit 888d4afbceb3d7d75906e6a4bfe009e07e5147f6 from
git@git.assembla.com:cavium/WindRiver.linux.git

When verifying PF_FUNC validity skip checking lf count for SSO block as
SSO can dynamically reconfigure lfs attached to the PF_FUNC.

Change-Id: I0225b384ba3993e6b3983f3cb704ad2f5b1d24cd
Signed-off-by: Pavan Nikhilesh <pbhagavatula@marvell.com>
Signed-off-by: Kevin Hao <kexin.hao@windriver.com>
---
 drivers/net/ethernet/marvell/octeontx2/af/rvu.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/net/ethernet/marvell/octeontx2/af/rvu.c b/drivers/net/ethernet/marvell/octeontx2/af/rvu.c
index ed4bef545624..228763a4f1c6 100644
--- a/drivers/net/ethernet/marvell/octeontx2/af/rvu.c
+++ b/drivers/net/ethernet/marvell/octeontx2/af/rvu.c
@@ -941,7 +941,7 @@ bool is_pffunc_map_valid(struct rvu *rvu, u16 pcifunc, int blktype)
 	pfvf = rvu_get_pfvf(rvu, pcifunc);
 
 	/* Check if this PFFUNC has a LF of type blktype attached */
-	if (!rvu_get_rsrc_mapcount(pfvf, blktype))
+	if (blktype != BLKTYPE_SSO && !rvu_get_rsrc_mapcount(pfvf, blktype))
 		return false;
 
 	return true;
-- 
2.31.1

